ui-core/src/components/inputs/checkbox/type.ts

```diff
@@ -0,0 +1,24 @@
+import { CheckboxProps } from './Checkbox';
+
+export type checkboxListItem = {
+  id: number;
+  parentId?: number; // Make parentId optional for root items
+  checkboxProps: CheckboxProps;
+};
+
+export enum CheckboxState {
+  UNCHECKED,
+  CHECKED,
+  INDETERMINATE,
+}
+
+export type ItemState = {
+  id: number;
+  state: CheckboxState;
+};
+
+export type checkboxTreeItem = {
```

Comment:
We use Pascal case for types
```suggestion
export type CheckboxTreeItem = {
```